cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocProvisioner.java

```diff
@@ -62,6 +66,7 @@
 
   // Key which is set to true if the instance only have private ip assigned to it else false
   private static final String PRIVATE_INSTANCE = "privateInstance";
+  private static final String BUCKET = "bucket";
```

Comment:
This is a static constant for the system property that comes from system context in case user has not provided bucket. Bucket is a optional property for dataproc cluster. We are following similar pattern.

Reply:
If user has not provided value, we will use checkpoint gcs bucket. We are using `BUCKET` property for that.
If user has provided bucket as property of provisioner, user provided bucket will be used.

Reply:
Got it.